File: Medium/2900. Longest Unequal Adjacent Groups Subsequence I.java
class Solution {
    public List<String> getWordsInLongestSubsequence(int n, String[] words, int[] groups) {
        List<String> ans = new ArrayList<>();
        for (int i = 0, b = -1; i < n; i++) {
            if (groups[i] != b) {
                b = groups[i];
                ans.add(words[i]);
            }
        }
        return ans;
    }
}

File: Easy/2900. Longest Unequal Adjacent Groups Subsequence I.py
from typing import List


class Solution:
    def getLongestSubsequence(self, words: List[str], groups: List[int]) -> List[str]:
        result = [words[0]]
        for i in range(1, len(groups)):
            if groups[i] != groups[i - 1]:
                result.append(words[i])
        return result